packages/paste-core/components/combobox/src/index.tsx

```diff
@@ -0,0 +1,147 @@
+  hasError?: boolean;
+  helpText?: string | React.ReactNode;
+  id?: string;
+  items: string[];
+  labelText: string | NonNullable<React.ReactNode>;
+  name?: string;
+  placeholder?: string;
+  required?: boolean;
+}
+
+const ComboboxIconWrapper: React.FC<{children: React.ReactNode}> = ({children}) => (
+  <Box
+    alignItems="center"
+    display="inline-flex"
+    position="absolute"
+    pointerEvents="none"
+    right="space40"
+    top="50%"
+    transform="translateY(-50%)"
+    zIndex="zIndex10"
+  >
+    {children}
+  </Box>
+);
+
+/* eslint-disable react/no-array-index-key */
+const Combobox: React.FC<ComboboxProps> = ({
+  autocomplete,
+  disabled,
+  helpText,
+  hasError,
+  id,
+  items,
+  labelText,
+  name,
+  placeholder,
+  required,
+}) => {
+  const [inputItems, setInputItems] = React.useState(items);
+  const {
+    getComboboxProps,
+    getInputProps,
+    getItemProps,
+    getLabelProps,
+    getMenuProps,
+    getToggleButtonProps,
+    highlightedIndex,
+    isOpen,
+  } = useComboboxPrimitive({
+    items: inputItems,
+    onInputValueChange: ({inputValue}) => {
+      if (autocomplete && inputValue) {
+        setInputItems(items.filter(item => item.toLowerCase().startsWith(inputValue.toLowerCase())));
+      }
+      return inputValue;
+    },
+  });
+  const helpTextId = useUID();
+  return (
+    <>
+      <FormLabel disabled={disabled} htmlFor={id} required={required} {...getLabelProps()}>
+        {labelText}
+      </FormLabel>
+      <Box {...getComboboxProps({role: 'combobox'})} display="flex" position="relative">
+        <FormInput
+          {...getInputProps()}
+          {...getToggleButtonProps({tabIndex: 0})}
+          disabled={disabled}
+          hasError={hasError}
+          id={id}
+          name={name}
+          placeholder={placeholder}
+          required={required}
+          type="text"
+        />
+        <ComboboxIconWrapper>
+          <ChevronDownIcon size="sizeIcon30" decorative aria-hidden="true" />
+        </ComboboxIconWrapper>
+      </Box>
+      <Box
+        {...getMenuProps()}
+        as="ul"
+        margin="space0"
+        padding="space0"
+        backgroundColor="colorBackgroundBody"
+        boxShadow="shadowCard"
+        borderBottomLeftRadius="borderRadius20"
+        borderBottomRightRadius="borderRadius20"
+        listStyleType="none"
+        overflow="hidden"
+      >
+        {isOpen &&
+          inputItems.map((item, index) => (
+            <Box
+              {...getItemProps({item, index})}
+              as="li"
+              backgroundColor={highlightedIndex === index ? 'colorBackgroundPrimaryLightest' : null}
+              key={`${item}${index}`}
+              padding="space30"
+              paddingLeft="space50"
+              paddingRight="space50"
+              _hover={{textDecoration: 'underline'}}
+            >
+              <Text as="span">{item}</Text>
+            </Box>
+          ))}
+      </Box>
+      {helpText && (
```

Comment:
Same here, help text is just composition.

Reply:
If we're keeping this in the control with Label, that's fine. Just wiring it up with id and aria-describedby on the input

Reply:
Added back in. Added the `aria-describedby`.